### PR TITLE
Remove bootstrap comps

### PR DESCRIPTION
**Pull request by Keelando**
Switched compilation Text and time slice toolbar over to Chakra. 

Compilation Text View refactored into Circuit, circuit after pauli, circuit after litinski boxes
- encountered difficulty vertically centering divs within boxes. It is most apparent for Circuit after Pauli rotations.
- looks "OK" as is but could be better, particularly for mobile and larger circuits.

Create temporary string parsing function to parse Compilation Text (will become JSON object from the lambda)

Create time slice stepper bar component
- this visually shows which time slice you are on rather than displaying a number.

### CODE REVIEW

**Comment by alexnguyenn** [comment]
Just did a check and seems like it crashed if I unselected Litinski Transform?

**Comment by Keelando** [comment]
> Just did a check and seems like it crashed if I unselected Litinski Transform?

Of course. Thanks, will fix

**Comment by Keelando** [comment]
> Just did a check and seems like it crashed if I unselected Litinski Transform?

This actually highlights a good reason to implement issue 7 on the lambda:
https://github.com/latticesurgery-com/compiler-lambda/issues/7

**Comment by alexnguyenn** [comment]
yeah i would say splitting compilation text should be lambda responsibility

**Comment by alexnguyenn** [comment]
> encountered difficulty vertically centering divs within boxes. It is most apparent for Circuit after Pauli rotations.

Have you tried [Center](https://chakra-ui.com/docs/layout/center)?

File: src/components/SliceIndexBar.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                            >
+                                <Center>
+                                    <Text>{index == selected ? index : null}</Text>
+                                </Center>
+                            </Box>
+                        )
+                    })}
+                </Flex>
+            </Box>
+        </>
```
Comment (alexnguyenn) [inline src/components/SliceIndexBar.tsx:34]: No need to use fragments here - remove `<></>`.

Also remove the commented code.

File: src/components/sections/LatticeView.tsx [hunk 2]
```diff
@@ -50,92 +65,110 @@ const LatticeView = ({ compilationResult }: LatticeViewProps): JSX.Element => {
     const handleChange = () => {
         setCompilationText(!showCompilationText)
     }
-
+    console.log(circuit_after_litinski)
```
Comment (alexnguyenn) [inline src/components/sections/LatticeView.tsx:68]: Remove `console.log`

File: src/components/sections/LatticeView.tsx [hunk 3]
```diff
@@ -50,92 +65,110 @@ const LatticeView = ({ compilationResult }: LatticeViewProps): JSX.Element => {
...
-            <div className="p-1 vertical-center">
-                <div>
-                    {showCompilationText ? (
-                        <div
-                            id="compilation-text"
-                            className="mb-3"
-                            css={css`
-                                margin-left: 10px;
-                            `}
+            {showCompilationText ? (
```
Comment (alexnguyenn) [inline src/components/sections/LatticeView.tsx:92]: I think it is a good idea to split the compilation text here to a separate component.

File: src/parseCompilationText.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    const compilation_text_split = compilation_text.split("Circuit")
+    const input_circuit = compilation_text_split[1].slice(2)
+
+    const pauli_rotations_split = compilation_text_split[2].split(":")
+    const compilationTextJson = {
+        input_circuit: input_circuit,
+        circuit_after_pauli_rotations: pauli_rotations_split[1].slice(1, -1),
+        circuit_after_litinski: "",
+    }
+    console.log("PIXZZA11111:", compilationTextJson)
```
Comment (alexnguyenn) [inline src/parseCompilationText.ts:12]: Remove `console.log`.

**Review by alexnguyenn — CHANGES_REQUESTED** [review]

**Comment by alexnguyenn** [comment]
I think the current implementation of timeslice is not ideal. See:

**Comment by alexnguyenn** [comment]
Also once all the bootstrap elements are removed, you can remove those 2 lines from `public/index.html` as well:

https://github.com/latticesurgery-com/web-ui/blob/13427d9b84b05d5dd0117bfe78b3f58a4ab3dfef/public/index.html#L16-L20

File: src/parseCompilationText.ts (hunk 4 above)
Comment (Keelando) [reply src/parseCompilationText.ts:12]: Thanks. That is embarrassing. Makes me think of how auto graders would give you zero points if your submission contained commented code, lol.

Comment (alexnguyenn) [reply src/parseCompilationText.ts:12]: haha i didn't notice those print statements at all until pressing F12 on my browser tho.

**Comment by Keelando** [comment]
> I think the current implementation of timeslice is not ideal. See:
> 
> image
> 
> We will definitely run into problems with longer circuit. I would suggest something like this: https://codesandbox.io/s/yupz1?file=/src/App.js:849-852, but if this is too difficult to implement then a simple `prev - current page number - next` design is ok.

What do you think about using something similar to what I have, but setting a maximum width/height such that the blocks shrink if more need to be rendered?

File: src/parseCompilationText.ts (hunk 4 above)
Comment (gwwatkin) [reply src/parseCompilationText.ts:12]: At least it's not obscene...

**Comment by gwwatkin** [comment]
@Keelando and @alexnguyenn  Really impressed by where you are taking this UI

File: src/parseCompilationText.ts (hunk 4 above)
Comment (Keelando) [reply src/parseCompilationText.ts:12]: PIZZA is my go to console.log. More Z's = more important.

**Comment by Keelando** [comment]
> The above screenshot was with the Bell pair example without Litinski applied. We can reach 100+ easily in the future.
> 
> > What do you think about using something similar to what I have, but setting a maximum width/height such that the blocks shrink if more need to be rendered
> 
> I am not sure wdym here but doesn't the problem remain? For longer circuits it will clutter the screen + with the shrinking it will make it impossible to interact with those block.

I can restrict the max width of the slice selector group, say 400px. Then I can shrink the width of each individual child block by dividing the max width by number of blocks. This will reduce the blocks a lot if there are 100. But if there are no gaps, any click will always result in selection of a slice. You just might hit 80 instead of 82, etc. 

If we are anticipating 100+ blocks, I would then propose using a volume slider knob:
https://chakra-ui.com/docs/form/slider

I can then map the slices to subsets of the slider range.

**Comment by alexnguyenn** [comment]
Slider is an interesting idea 🤔. Tho I think it would be out of the scope for this PR - I do want to close #98 first.

File: src/components/sections/LatticeView.tsx (hunk 3 above)
Comment (Keelando) [reply src/components/sections/LatticeView.tsx:92]: I agree. A formal component will be good. I have a PR in the works on the main repo to send the data as a formatted JSON object

**Comment by Keelando** [comment]
The slice selector really is just a placeholder, just experimenting with what is possible. Happy to have it torn down and rebuilt better.

The compilation text components need to be formalized, again just a placeholder.

It appears the lattice view still requires bootstrap and font awesome because when I removed it, the lattice didnt render correctly. 

So to fully remove the dependency we will have to rebuild the lattice view with Chakra

**Comment by gwwatkin** [comment]
> So to fully remove the dependency we will have to rebuild the lattice view with Chakra
Don't know who we got there :sweat_smile:

**Comment by Keelando** [comment]
> > So to fully remove the dependency we will have to rebuild the lattice view with Chakra
> 
> Don't know who we got there :sweat_smile: 

I'm pretty sure I tinkered with it at some point and then gave up. With a new framework (Chakra) I'm sure Alex and I can come up with something good.

**Comment by alexnguyenn** [comment]
image

I think you can remove this button also.

**Review by gwwatkin — APPROVED** [review]
Lets merge this for now, so the lattice view looks better for the UF meeting. @alexnguyenn  what do you think?

As for the slice selector, the current design is ok for short computations. For longer computations, I'm fine with either pagination or sliders, as long as you can still see the number and still have a fine grained +1/-1 navigation option

**Review by alexnguyenn — APPROVED** [review]
Agree with @gwwatkin. Feel free to merge this in (maybe after removing the new circuit button i mentioned above).

**Pull request merged**